craft_store/base_client.py

```diff
@@ -298,6 +299,21 @@ def list_revisions(self, name: str) -> List[RevisionModel]:
 
         return [RevisionModel.unmarshal(r) for r in response["revisions"]]
 
+    def list_resource_revisions(
+        self, name: str, resource_name: str
+    ) -> List[CharmResourceRevision]:
+        """List the revisions for a specific resource of a specific name."""
+        namespace = self._endpoints.namespace
+        if namespace != "charm":
+            raise NotImplementedError(
+                f"Cannot get resource revisions in namespace {namespace}."
+            )
+        endpoint = f"/v1/{namespace}/{name}/resources/{resource_name}/revisions"
+        response = self.request("GET", self._base_url + endpoint)
+        model = response.json()
+
+        return [CharmResourceRevision.unmarshal(r) for r in model["revisions"]]
```

Comment:
do you think there's a risk of charmhub giving us bad data? For example, a list of architectures with repeated values? I ask this because if there's a problem in a project yaml it's correct to fail immediately, but I don't know what our "philosophy" when handling bad data from a store is/should be.
(this doesn't block the review I was just curious)